R/zchunk_L141.hfc_R_S_T_Y.R

```diff
@@ -51,67 +48,210 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
+      mutate(share = value / total) %>%
+      select(GCAM_region_ID,year,service,share,value)
+
+    # Add res/com cooling share to HFC and make adjustment
+    L141.hfc_R_S_T_Yh_share<- left_join(L141.hfc_R_S_T_Yh.long,
+                                        L141.R_cooling_T_Yh.long ,
+                                        by = c("GCAM_region_ID", "year", "supplysector" = "service")) %>% # there should be NAs.
+                                        #there are sectors that don't have emissions
+      replace_na(list(share=1)) %>% # replace those shares with "1"
+      mutate(emissions = emissions * share)
+
+    # Scale HFC Emissions to SSP2 HFC (Guus Velders) inventory (year 2010), where available
+
+    # EDGAR data doesn't have 2010 data, so match EDGAR 2008 data with GV 2010 data
+    # Duplicate EDGAR 2008 data and rename as 2010
+    TEMP <- subset(L141.hfc_R_S_T_Yh_share, year == 2008)
+    TEMP$year <- 2010
+    L141.hfc_R_S_T_Yh_share <- bind_rows(L141.hfc_R_S_T_Yh_share, TEMP)
+
+    # Process SSP2 HFC (Guus Velders) inventory
+    HFC_Inventory_GV <- HFC_Inventory_GV %>% mutate(Species = gsub("-","", Species)) %>%
+      mutate(Species = gsub("4310mee", "43",Species)) %>%
+      rename(SSP2_tot = Emissions)
+
+    # Add SSP2 values to EDGAR HFC emissions then calculate scalar
+    L141.hfc_scaler <- L141.hfc_R_S_T_Yh_share %>% group_by(year,Non.CO2) %>%
+      summarize(EDGAR_tot = sum(emissions))%>%
+      arrange(Non.CO2,year) %>%
+      left_join(HFC_Inventory_GV, by = c(year = "Year", Non.CO2 = "Species")) %>% # some entries not in GUUS data, default to scaler =1
+      mutate(scaler = SSP2_tot / EDGAR_tot) %>%
+      replace_na(list(scaler = 1)) %>%
+      mutate_at(vars(), funs(replace(., is.infinite(.), 1))) %>% # replace na and Infinite values with 1.
+        # If a real scalar isn't calculated (ie: no GV value or EDGAR = 0) to match to GV data, keep the EDGAR value.
+      select(year, Non.CO2, scaler)
+
+    # Add scalar to HFC emissions, calculate adjusted emissions
+    L141.hfc_R_S_T_Yh <- L141.hfc_R_S_T_Yh_share %>% left_join(L141.hfc_scaler,
+                                        by = c("year", "Non.CO2")) %>% # there should be NA's, as not all sectors have emissions/scalers
+      mutate(adj_emissions = emissions * scaler) %>%
+      group_by(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2, year) %>%
+      summarise(adj_emissions = sum(adj_emissions, na.rm = T)) %>%
+      replace_na(list(adj_emissions = 0)) %>%  #replace NAs with zero. Some sectors have zero emissions.
+      rename(value = adj_emissions)
+
+    # Compute final cooling HFC emissions factors
+    L141.hfc_ef_R_cooling_Yh <- L141.hfc_R_S_T_Yh %>%
+      filter(supplysector %in% c("comm cooling", "resid cooling"), year %in% HISTORICAL_YEARS) %>%
+      left_join_error_no_match(L141.R_cooling_T_Yh.long %>% select(GCAM_region_ID, year,service,value) %>% rename(energy = value),
```

Comment:
FYI could just do `select(GCAM_region_ID, year, service, energy = value)`. I'll make this change.